bigchaindb/tendermint/lib.py

```diff
@@ -23,23 +23,29 @@
 
 class BigchainDB(Bigchain):
 
-    def post_transaction(self, transaction):
+    def post_transaction(self, transaction, mode):
         """Submit a valid transaction to the mempool."""
+        mode_list = ('broadcast_tx_async',
+                     'broadcast_tx_sync',
+                     'broadcast_tx_commit')
+        if not mode or mode['mode'] not in mode_list:
+            raise ValidationError(('Mode must be one of the following {}.')
+                                  .format(', '.join(mode_list)))
 
         payload = {
-            'method': 'broadcast_tx_async',
+            'method': mode['mode'],
             'jsonrpc': '2.0',
             'params': [encode_transaction(transaction.to_dict())],
             'id': str(uuid4())
         }
         # TODO: handle connection errors!
         requests.post(ENDPOINT, json=payload)
 
-    def write_transaction(self, transaction):
+    def write_transaction(self, transaction, **kwargs):
```

Comment:
The function is being called by unpacking the args, so `mode` be directly extracted here right?